### ISSUE TITLE
Fresh Install longhorn volumes not getting attached to pods

### ISSUE THREAD

**Issue by rajivml** [longhorn/longhorn#2241]
**Describe the bug**

I have created an 3 node RKE2 cluster on bare metal azure VM's and I have installed longhorn as per the documentation through kubectl method

Each machine has one 256 GB SSD attached to it and this is how I have partitioned each disk and attached the same via longhorn UI

Now to testing the working of it , I have created an volume as per the documentation 
kubectl create -f https://raw.githubusercontent.com/longhorn/longhorn/v1.0.0/examples/pod_with_pvc.yaml

but the PVC just got stuck in attached state, it's getting attached and detached again, Attaching -> Attached -> Detached -> Attaching -> looping forever 

```
mkfs -t ext4 /dev/sdc
mkdir -p /mnt/sdc
mount /dev/sdc /mnt/sdc
```

```
curl -sfL https://get.rke2.io --output install.sh
chmod +x install.sh
INSTALL_RKE2_CHANNEL=latest ./install.sh

cat /etc/rancher/rke2/config.yaml

selinux: false
server: https://xx.xx.122.14:9345
token: 
ae70e::server:0a83109039dc25b19b2b16481a419967
tls-san:
  - xx.xx.122.14
```

![image](https://user-images.githubusercontent.com/46923567/107120487-e546f700-68b3-11eb-9a2c-d530242f39cd.png)

![image](https://user-images.githubusercontent.com/46923567/107120503-fe4fa800-68b3-11eb-8327-74604131a708.png)

![image](https://user-images.githubusercontent.com/46923567/107121434-62746b00-68b8-11eb-867b-74543516caf2.png)

**Expected behavior**
PVC should get attached to the pod

Please find the support bundle attached with the ticket.

**Environment:**
 - Longhorn version: Tried both 1.0.0 and 1.1.1
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: RKE2
   - Number of management node in the cluster: 3
   - Number of worker node in the cluster: 0
 - Node config
   - OS type and version: RHEL 8.2
   - CPU per node: 8 Core
   - Memory per node: 32
   - Disk type(e.g. SSD/NVMe): SSD
   - Network bandwidth between the nodes: All Nodes are within the same VPC
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): Azure VM's
 - Number of Longhorn volumes in the cluster: 3

Please find the attached support bundle for both 0.8.0 and 1.1.1 versions and please let us know how to get it working 

[longhorn-support-bundle_3c9f69e8-1d79-4735-8296-b0c85dbd53c4_2021-02-06T14-46-50Z.zip](https://github.com/longhorn/longhorn/files/5937013/longhorn-support-bundle_3c9f69e8-1d79-4735-8296-b0c85dbd53c4_2021-02-06T14-46-50Z.zip)
[longhorn-support-bundle_9909a827-5083-43ed-87dd-83d798eb91f5_2021-02-06T13-17-31Z.zip](https://github.com/longhorn/longhorn/files/5937014/longhorn-support-bundle_9909a827-5083-43ed-87dd-83d798eb91f5_2021-02-06T13-17-31Z.zip)

**Comment by rajivml** [comment]
```
Events:
  Type     Reason            Age   From               Message
  ----     ------            ----  ----               -------
  Warning  FailedScheduling  29s   default-scheduler  0/3 nodes are available: 3 pod has unbound immediate PersistentVolumeClaims.
  Warning  FailedScheduling  29s   default-scheduler  0/3 nodes are available: 3 pod has unbound immediate PersistentVolumeClaims.
  Normal   Scheduled         23s   default-scheduler  Successfully assigned default/volume-test to rke2-3
[root@rke2-3 longhorn]#
[root@rke2-3 longhorn]#
[root@rke2-3 longhorn]#
[root@rke2-3 longhorn]# kubectl get pvc
NAME                STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
longhorn-volv-pvc   Bound    pvc-657777db-3829-42ce-ad1b-fccf30c3f425   2Gi        RWO            longhorn       40s
[root@rke2-3 longhorn]# kubectl get pv
NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                       STORAGECLASS   REASON   AGE
pvc-657777db-3829-42ce-ad1b-fccf30c3f425   2Gi        RWO            Delete           Bound    default/longhorn-volv-pvc   longhorn                43s
```

**Comment by shuo-wu** [comment]
One quick question, have you installed iscsi on all of your worker nodes? https://longhorn.io/docs/0.8.0/install/requirements/

To debug the issue, we need to check the info of the stuck pod `volume-test` (by using `kubectl describe`), and the logs of longhorn manager pods or instance manager pods (by using `kubectl logs`).

**Comment by rajivml** [comment]
yes, I have installed yum install iscsi-initiator-utils on all the nodes and I have verified the setup through the shell script that has been documented

To share the logs, I no longer have that environment, but with the ticket, I have attached the support dump, wouldn't it contain the logs that are required to debug the issue ?

**Comment by shuo-wu** [comment]
1. I see tons of the iscsi related errors in your cluster. Can you check if the iscsi service is started among all of your nodes?:
```
2021-02-06T14:37:59.604753678Z time="2021-02-06T14:37:59Z" level=warning msg="pvc-97ad64ce-99b2-4bc0-ac52-e8ae30b4b568-e-0220a350: time=\"2021-02-06T14:37:55Z\" level=warning msg=\"FAIL to discover due to Failed to execute: nsenter [--mount=/host/proc/1/ns/mnt --net=/host/proc/1/ns/net iscsiadm -m discovery -t sendtargets -p 10.42.2.14], output , stderr, Failed to connect to bus: No data available\\niscsiadm: can not connect to iSCSI daemon (111)!\\nFailed to connect to bus: No data available\\niscsiadm: can not connect to iSCSI daemon (111)!\\niscsiadm: Cannot perform discovery. Initiatorname required.\\niscsiadm: Could not perform SendTargets discovery: could not connect to iscsid\\n, error exit status 20\""
2021-02-06T14:37:59.604781179Z time="2021-02-06T14:37:59Z" level=warning msg="pvc-97ad64ce-99b2-4bc0-ac52-e8ae30b4b568-e-0220a350: time=\"2021-02-06T14:37:55Z\" level=warning msg=\"Nodes cleaned up for iqn.2019-10.io.longhorn:pvc-97ad64ce-99b2-4bc0-ac52-e8ae30b4b568\""
2021-02-06T14:37:59.604799079Z time="2021-02-06T14:37:59Z" level=warning msg="pvc-97ad64ce-99b2-4bc0-ac52-e8ae30b4b568-e-0220a350: time=\"2021-02-06T14:37:58Z\" level=error msg=\"Failed to startup frontend: Failed to execute: nsenter [--mount=/host/proc/1/ns/mnt --net=/host/proc/1/ns/net iscsiadm -m node -T iqn.2019-10.io.longhorn:pvc-97ad64ce-99b2-4bc0-ac52-e8ae30b4b568 -p 10.42.2.14 --login], output , stderr, iscsiadm: No records found\\n, error exit status 21\""
2021-02-06T14:37:59.604803279Z time="2021-02-06T14:37:59Z" level=warning msg="pvc-97ad64ce-99b2-4bc0-ac52-e8ae30b4b568-e-0220a350: 2021/02/06 14:37:58 failed to start up frontend: Failed to execute: nsenter [--mount=/host/proc/1/ns/mnt --net=/host/proc/1/ns/net iscsiadm -m node -T iqn.2019-10.io.longhorn:pvc-97ad64ce-99b2-4bc0-ac52-e8ae30b4b568 -p 10.42.2.14 --login], output , stderr, iscsiadm: No records found"
```

2. I don't know why there are volumes using the non-existing v0.8.0 engine image. And I am not sure if this old engine image is still compatible in v1.1.0. If possible, can you upgrade them to the default engine image?
```
2021-02-06T14:38:29.202969774Z time="2021-02-06T14:38:29Z" level=warning msg="Error syncing Longhorn replica longhorn-system/pvc-02ec116d-889a-48de-8376-e54c95c1fe9f-r-5d7abbb2: fail to sync replica for longhorn-system/pvc-02ec116d-889a-48de-8376-e54c95c1fe9f-r-5d7abbb2: failed to cleanup the related replica process before deleting replica pvc-02ec116d-889a-48de-8376-e54c95c1fe9f-r-5d7abbb2: failed to get engine image object based on image name longhornio/longhorn-engine:v0.8.0: engineimage.longhorn.io \"ei-e10d6bf5\" not found"
```

**Comment by joshimoo** [comment]
possibly related to iscsci installation on redhat #1741

**Comment by joshimoo** [comment]
This user found that they are missing a library on rhel, might help with open-iscsi problems.
#2163

**Comment by shuo-wu** [comment]
Based on our QA test result in RHEL8.3, RHEL 8.3 will not start iscsid service automatically then lead to errors like `iscsiadm: No records found` and the users need to start it manually. Can you check if there is a similar case in RHEL8.2?

**Comment by rajivml** [comment]
I have observed this on few installs where it started automatically but on few it didn't , it's very random

**Comment by yasker** [comment]
We will check back this issue once we're done with https://github.com/longhorn/longhorn/issues/1741

**Comment by innobead** [comment]
@khushboo-rancher or @cclhsu I assume this should be already tackled in https://github.com/longhorn/longhorn-manager/pull/840 because of the prerequisite installation already verified in CentOS 8.x. Can you double confirm? Thanks.

**Comment by cclhsu** [comment]
The logs which user provided are the same as described in https://github.com/longhorn/longhorn/issues/2319.  Solution was tested on Centos 7.8 and 8.2.  @kaskavalci Can you confirm if it is the same for RHEL 8.2?

**Comment by khushboo-rancher** [comment]
https://github.com/longhorn/longhorn/issues/2319 is validated with RHEL7.8 and RHEL8.2

**Comment by innobead** [comment]
Closing, resolved by #2319
